projects/plugins/boost/app/modules/critical-css/class-critical-css.php

```diff
@@ -652,31 +654,76 @@ public function get_critical_css() {
 	 * @see style_loader_tag
 	 */
 	public function asynchronize_stylesheets( $html, $handle, $href, $media ) {
+		// If is AMP, do not alter the stylesheet loading.
 		if ( function_exists( 'is_amp_endpoint' ) && is_amp_endpoint() ) {
 			return $html;
 		}
 
+		// If there is no critical CSS, do not alter the stylesheet loading.
 		if ( false === $this->get_critical_css() ) {
 			return $html;
 		}
 
-		if ( ! apply_filters( 'jetpack_boost_async_style', true, $handle ) ) {
+		$available_methods = array(
+			'instant'  => 'media="' . $media . '"',
+			'async'    => 'media="not all" onload="this.media=\'all\'"',
+			'deferred' => 'media="not all"',
+		);
+
+		/**
+		 * Loading method for stylesheets.
+		 *
+		 * Filter the loading method for each stylesheet for the screen with following values:
+		 *     instant  - Stylesheet loading behaviour is not altered. Stylesheet loading is instant and the process blocks the page rendering.
+		 *     async    - Stylesheets are loaded asynchronously. Styles are applied once the stylesheet is loaded completely without render blocking.
+		 *     deferred - Loading of stylesheets are deferred till the window load event. Styles from all the stylesheets are applied at once after the page load.
+		 *
+		 * @see onload_flip_stylesheets for how stylesheets loading is deferred.
+		 *
+		 * @param string $handle The style's registered handle.
+		 * @param string $media  The stylesheet's media attribute.
+		 *
+		 * @todo  Retrieve settings from database, either via auto-configuration or UI option.
+		 */
+		$method = apply_filters( 'jetpack_boost_async_style', 'async', $handle, $media );
+
+		// If the method is explicitly false, do not load the specific stylesheet.
+		if ( ( false === $method ) ) {
+			return false;
+		}
+
+		// If the loading method is not allowed or 'instant', do not alter the stylesheet loading.
+		if ( ! isset( $available_methods[ $method ] ) || ( 'instant' === $method ) ) {
 			return $html;
 		}
-		$async_media = apply_filters( 'jetpack_boost_async_media', array( 'all', 'screen' ) );
 
-		// Convert stylesheets intended for screens.
-		if ( in_array( $media, $async_media, true ) ) {
-			$media_replacement = 'media="not all" onload="this.media=\'all\'"';
-			$html_no_script    = '<noscript>' . $html . '</noscript>';
-			$html              = preg_replace( '~media=[\'"]?[^\'"\s]+[\'"]?~', $media_replacement, $html );
+		$html_no_script = '<noscript>' . $html . '</noscript>';
+
+		// Update the stylesheet markup for allowed methods.
+		$html = preg_replace( '~media=[\'"]?[^\'"\s]+[\'"]?~', $available_methods[ $method ], $html );
```

Comment:
Hmm.. This is an interesting problem. If the value is null and we do not alter the $html, `asynchronize_stylesheets` will not do what it is intended to do - silently. However, not checking for null is even worse since we simply return null instead of real $html.

Reply:
After some research, I found that null return is an extremely rare case. One example is here: https://www.php.net/manual/en/function.preg-replace.php#84285. I think it is not worth adding this check.

Reply:
Makes sense. 👍